test/simple/test-net-create-connection.js

```diff
@@ -38,19 +38,34 @@ server.listen(tcpPort, 'localhost', function() {
     ++clientConnected;
   }
 
+  function fail(opts, errtype, msg) {
+    assert.throws(function() {
+      var client = net.createConnection(opts, cb);
+    }, function (err) {
+      return err instanceof errtype && msg === err.message;
+    });
+  }
+
   net.createConnection(tcpPort).on('connect', cb);
   net.createConnection(tcpPort, 'localhost').on('connect', cb);
   net.createConnection(tcpPort, cb);
   net.createConnection(tcpPort, 'localhost', cb);
 
-  assert.throws(function () {
-    net.createConnection({
-      port: 'invalid!'
-    }, cb);
-  });
+  fail({
+    port: 'invalid!'
```

Comment:
I think it can't hurt to add more tests with that cover the rest of invalid values such as `null`, `NaN`, etc. What do you think?